src/ToCore/Type.ml

```diff
@@ -34,11 +34,9 @@ let pack_type (type k) (k : S.kind) (tp : k T.typ) : T.Type.ex =
 let rec tr_effrow_end env (eff : S.Type.effrow_end) =
   match eff with
   | EEClosed -> T.TEffPure
-  | EEUVar _  ->
-    (* TODO: they can be supported, and its especially useful in REPL *)
-    InterpLib.Error.report ~cls:FatalError
-      "Unsolved unification variables left.";
-    raise InterpLib.Error.Fatal_error
+  | EEUVar (_,uvar) ->
+    let uid = Lang.Unif.UVar.uid uvar in
```

Comment:
In this place we have to options. Either we are in the REPL mode, then we translate the variable as here. Or we are in the batch mode, then we should raise an error.